File: client/src/models/appointmentOutDTO.ts
import { PrescriptionDetailOutDTO } from "./prescriptionDetailOutDTO";
import { VisitType } from "./visitType";

export interface AppointmentOutDTO {
  id: number;
  patientId: number;
  patientName: string;
  doctorId: number;
  doctorName: string;
  appointmentDate: string; 
  diagnosis: string;
  notes: string;
  visitType: VisitType;
  prescriptionDetails: PrescriptionDetailOutDTO[];
}

File: backend/backend.Application/Models/AppointmentOutDTO.cs
using backend.Domain.Enums;

namespace backend.Application.Models
{
    public class AppointmentOutDTO
    {
        public int Id { get; set; }
        public int PatientId { get; set; }
        public string PatientName { get; set; } = null!;
        public int DoctorId { get; set; }
        public string DoctorName { get; set; } = null!;
        public DateTime AppointmentDate { get; set; }
        public string Diagnosis { get; set; } = string.Empty;
        public string Notes { get; set; } = string.Empty;
        public VisitType VisitType { get; set; }
        public List<PrescriptionDetailOutDTO> PrescriptionDetails { get; set; } = new();
    }
}